Source/WebKit/UIProcess/wpe/AcceleratedBackingStoreDMABuf.cpp

```diff
@@ -133,9 +133,28 @@ void AcceleratedBackingStoreDMABuf::frame(uint64_t bufferID)
         return;
     }
 
+    WebCore::Damage::Rects damageRects;
+    const WPERectangle* rects;
+    guint nRects;
+
+    if (damage.isInvalid()) {
+        rects = nullptr;
+        nRects = 0;
+    } else if (damage.isEmpty()) {
+        static const WPERectangle emptyRect = { 0, 0, 0, 0 };
+        rects = &emptyRect;
+        nRects = 1;
+    } else {
+        static_assert(sizeof(WebCore::IntRect) == sizeof(WPERectangle));
+        damageRects = damage.rects();
+        ASSERT(damageRects.size() <= std::numeric_limits<guint>::max());
+        rects = reinterpret_cast<const WPERectangle*>(damageRects.data());
+        nRects = damageRects.size();
+    }
```

Comment:
I think we could use Vector<IntRect, 1> for all the cases. When invalid we don't need to do anything, when empty just append({ }).